int M1_dir
int M2_dir
int M1_Speed
int M2_Speed